### PR TITLE
Add docs for user import

### PR DESCRIPTION
**Pull request by hortha**
### Context

Need to document user import process to help support it

### Changes proposed in this pull request

New document and diagram explaining user import process

### Guidance to review

Added new markdown document + a few tweaks to code that I noticed while writing the document.

### Checklist

-   [x] Attach to Trello card
-   [x] Rebased master
-   [x] Cleaned commit history
-   [x] Tested by running locally
-   [x] Reminder created to manually clean any removed app settings post deployment

### CODE REVIEW

**Comment by hortha** [comment]
[Document Identity user import logic](https://trello.com/c/DHl36MYY/681-document-identity-user-import-logic)

File: docs/user-import.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+## Overview
+
+As more and more DfE services migrate to using `Identity` to manage authentication and authorisation, there is the need to migrate existing users of these services into identity.  
+While this could be done the next time the user interacts with the specific service, it would mean them having to re-enter their details in order to create a teaching account.  
+The user import feature has been created to automate the creation of teaching accounts in identity from users of other DfE services as much as possible.
+
+## Import File Definition
+
+The user import requires a CSV file with a header with the following fields:
```
Comment (gunndabad) [inline docs/user-import.md:11]: Worth calling out that all fields are mandatory except TRN?

File: docs/user-import.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+| ------------------- | ------------------------------------------------------------ | -----------------------------------------------|
+| RowNumber           | The row number from the original uploaded CSV file           | An integer                                     |
+| Id                  | The unique ID associated with the user in the source service | A string                                       |
+| UserId              | The unique ID associated with the user in identity           | A GUID                                         |
+| UserImportRowResult | The outcome associated with the row of data from the CSV     | One of `None`, `UserAdded`, `UserUpdated` or `Invalid` |
+| Notes               | Any notes e.g. errors                                        | A string with multiple notes separated by ". " |
+| RawData             | The raw row of data from the original uploaded CSV file      | A string                                       |
+
+This CSV can then be used by the source service to enhance it's data with user IDs from Identity
```
Comment (gunndabad) [inline docs/user-import.md:37]:
```suggestion
This CSV can then be used by the source service to enhance its data with user IDs from Identity.
```

File: docs/user-import.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+| DATE_OF_BIRTH | The user's date of birth                                     | A valid date in ddMMyyyy format e.g. 03051971      |
+| TRN           | The user's TRN (if known in the source service)              | Empty or a 7 digit number                          |
+
+\* Note that there is no validation of whether the email address supplied is actually a valid personal email
+
+## Download File Definition
+
+The results of each file import can be downloaded in a CSV file with the following fields:
+
+| Header Name         | Description                                                  | Format                                         |
```
Comment (gunndabad) [inline docs/user-import.md:28]: Now I'm seeing it in doc form, I wonder whether we should change the header format out this output file. Seems a little strange to have ONE_CASING for the import and AnotherCasing for the output.

Comment (hortha) [reply docs/user-import.md:28]: Could change the output one maybe ?

Comment (hortha) [reply docs/user-import.md:28]: Funny I sort of thought the same

File: docs/user-import.md (hunk 1 above)
Comment (hortha) [reply docs/user-import.md:11]: Yeah I thought I alluded to that with the "Empty" being a valid option but .. and technically you still need it in the CSV even if it's empty

File: docs/user-import.md (hunk 3 above)
Comment (hortha) [reply docs/user-import.md]: I've fixed this and updated the docs

File: docs/user-import.md (hunk 2 above)
Comment (hortha) [reply docs/user-import.md]: Done

File: docs/user-import.md (hunk 1 above)
Comment (hortha) [reply docs/user-import.md:11]: Added a column to the table to indicate the mandatoryness of the fields

**Review by gunndabad — APPROVED** [review]

**Pull request merged**